From d02449c48d48bda89bc68a4db8aa2dac7a601f6b Mon Sep 17 00:00:00 2001
From: Fancy Fang <chen.fang@nxp.com>
Date: Mon, 7 Aug 2017 19:40:14 +0800
Subject: [PATCH 2338/5242] MLK-16136-17 video: fbdev: dcss: clear global
 alpha in blanking one channel

commit  76aa4be21e9981709058adfa9b63a4b7c7c12280 from
https://source.codeaurora.org/external/imx/linux-imx.git

The global alpha for one channel should be cleared when
blanding this channel.

Signed-off-by: Fancy Fang <chen.fang@nxp.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 drivers/video/fbdev/mxc/imx_dcss.c |    6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/drivers/video/fbdev/mxc/imx_dcss.c b/drivers/video/fbdev/mxc/imx_dcss.c
index e705056..6330a07 100644
--- a/drivers/video/fbdev/mxc/imx_dcss.c
+++ b/drivers/video/fbdev/mxc/imx_dcss.c
@@ -2761,6 +2761,12 @@ static int dcss_channel_blank(int blank,
 	case FB_BLANK_POWERDOWN:
 		/* disable dtg */
 		dtg_ctrl &= ~(0x1 << (2 - cinfo->channel_id));
+
+		/* clear global alpha */
+		if (cinfo->channel_id == DCSS_CHAN_MAIN)
+			dtg_ctrl &= ~(0xff << 24);
+		else
+			dtg_ctrl |= (0xff << 24);
 		break;
 	default:
 		return -EINVAL;
-- 
1.7.9.5

